pkg/actor/cluster_restart.go

```diff
@@ -185,9 +190,10 @@ func (r *clusterRestart) rollingSts(ctx context.Context, sts *appsv1.StatefulSet
 			return errors.Wrapf(err, "error rolling update stategy on pod %d", int(partition))
 		}
 
-		// wait 1 minute between updates
-		l.V(int(zapcore.DebugLevel)).Info("sleeping", "duration", sleepDuration.String(), "label", "between restarting pods")
-		time.Sleep(sleepDuration)
+		// wait for all replicas to be up
+		if err := healthChecker.Probe(ctx, l, "between restarting pods", int(partition)); err != nil {
```

Comment:
We should maybe add an issue to add the `healthchecker` in the full restart as well. There we only wait on the `scale.WaitUntilStatefulSetIsReadyToServe` to finish. The DB is unavailable for that but it might be valuable to ensure everything is good before finishing the op. wdyt @keith-mcclellan?

@alinadonisa the above does not need to be a part of this PR though

Reply:
fullRestart actually restarts all the nodes at once - it's intended largely to handle cases where we know we need to take an outage (like changing CA's for the node cert) to fix something. That action will always cause database unavailability so I'm not sure what adding this to that restart would do @udnay ?  The only utility I could see is it could be used to block other admin actions by the CR until the database is completely healthy, but I'd think we would fix that in the refactor using an admission webhook rather than continuing to enhance this restart actor.